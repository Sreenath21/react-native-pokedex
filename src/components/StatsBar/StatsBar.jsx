import React from 'react';
import {View, Text} from 'react-native';

import Progress from '../Progress/Progress';
import styles from './StatsBar.style';

const StatsBar = ({name, value}) => {
  return (
    <View style={styles.container}>
      <View style={styles.stat}>
        <Text style={styles.text}>{name}</Text>
      </View>
      <Progress value={value} />
    </View>
  );
};

export default StatsBar;
