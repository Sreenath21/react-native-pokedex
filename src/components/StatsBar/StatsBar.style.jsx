import {StyleSheet} from 'react-native';
import colors from '../../utils/colors';

export default StyleSheet.create({
  container: {
    marginTop: 16,
    flexDirection: 'row',
    columnGap: 10,
    alignItems: 'center',
  },
  stat: {
    flex: 3,
  },
  text: {
    color: colors.primary,
  },
});
