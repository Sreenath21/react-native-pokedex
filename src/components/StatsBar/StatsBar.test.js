import React from 'react';
import StatsBar from './StatsBar';
import renderer from 'react-test-renderer';

it('StatsBar component renders correctly', () => {
  const tree = renderer.create(<StatsBar name="Height" value={80} />);
  expect(tree).toMatchSnapshot();
});
