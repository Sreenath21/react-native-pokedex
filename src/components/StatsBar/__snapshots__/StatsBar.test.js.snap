// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsBar component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "columnGap": 10,
      "flexDirection": "row",
      "marginTop": 16,
    }
  }
>
  <View
    style={
      {
        "flex": 3,
      }
    }
  >
    <Text
      style={
        {
          "color": "#391363",
        }
      }
    >
      Height
    </Text>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "#9fc6c7",
          "flex": 10,
          "height": 12,
          "overflow": "hidden",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "#391363",
            "flex": 1,
            "height": 12,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          },
          {
            "width": "80%",
          },
        ]
      }
    />
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 10,
          "fontWeight": 500,
          "marginLeft": 5,
        }
      }
    >
      80
    </Text>
  </View>
</View>
`;
